index.html

```diff
@@ -406,25 +468,195 @@ <h3 id="installation-sec">
     </section>
     <section>
       <h2>
-        Installation Events
+        Installation API and events
       </h2>
+      <p>
+        DOM events <a>fired</a> by this specification use the <dfn>application
+        life-cycle task source</dfn>.
+      </p>
+      <section data-dfn-for="AppBannerPromptOutcome">
+        <h3>
+          <code>AppBannerPromptOutcome</code> enum
+        </h3>
+        <pre class="idl">
+        enum AppBannerPromptOutcome {
+          "accepted",
+          "dismissed"
+        };
+      </pre>
+        <p>
+          The <dfn>AppBannerPromptOutcome</dfn> represent the possibly outcomes
+          from the end-user being <a data-lt=
+          "presents an install prompt">presented with an with an automated
+          install prompt</a>.
+        </p>
+      </section>
+      <section data-dfn-for="BeforeInstallPromptEvent">
+        <h3>
+          <code>BeforeInstallPromptEvent</code> Interface
+        </h3>
+        <pre class="idl">
+          [Constructor(DOMString typeArg, optional BeforeInstallPromptEventInit eventInit)]
+          interface BeforeInstallPromptEvent : Event {
+              readonly attribute Promise&lt;AppBannerPromptOutcome&gt; userChoice;
+              Promise&lt;void&gt; prompt();
+          };
+
+          dictionary BeforeInstallPromptEventInit : EventInit {
+            AppBannerPromptOutcome userChoice;
```

Comment:
There was a discussion about this on the previous review round that went unfinished:

[My previous statement](https://github.com/w3c/manifest/pull/506/files/4af07193642bf9ae5d3f002337c64b196b0db908#r84207025)

I still don't see the point of this `userChoice` constructor argument being not a promise, implying that you've already made the choice and now you only have to understand why*. It _might_ be useful if `userChoice` was a promise, allowing the person fabricating this event to allow the choice to be resolved at some future point. But even then, I see little point in actually supporting the event fabrication use case at all. This would all (speccing, implementation, testing) be a lot easier if we banned fabrication altogether, or if the Web Platform doesn't allow that, just letting it exist but not do much (and not specifically inventing features to support this useless use case).

If you can justify why someone (a web developer, not a UA implementor) would want to create a `BeforeInstallPromptEvent` with a `userChoice` -- even a promise -- then I'm all ears.

\* Bonus points if you get that reference.